src/ipc/local/LocalIPCRequestHandler.h

```diff
@@ -61,9 +60,10 @@ class FaceliftIPCLocalLib_EXPORT LocalIPCRequestHandler
 {
 
 public:
-    virtual ~LocalIPCRequestHandler() = default;
-    virtual void deserializePropertyValues(LocalIPCMessage &msg, bool isCompleteSnapshot) = 0;
-    virtual void deserializeSignal(LocalIPCMessage &msg) = 0;
+    virtual void unmarshalProperties(const QVariantMap& changedProperties) = 0;
```

Comment:
no QVariant in local IPC

Reply:
So let's look at what QVariant replaces. QVariant replaces the "facelift serialization" mechanism which in first place was reinvention of the wheel. You don't need to serialize/deserialize properties if you need only to treat them in the same process generically. 
Anyhow in terms of difference now we are looking at cost of serializing a strut (facelift serialization) vs cost of cloning it (QVariant) and if the latter deems to be more costly then the former it again can be traced back to wrong fundamentals of facelift::structure class and overhead of initialization std::tuples.  
So "no QVariant in local IPC" is not a technical review comment.
If you want to do facelift any good I suggest we rethink the facelift::structure.

Reply:
I do not understand what the point about std::tuples has to do with that discussion. Even if we were using "plain" data members instead of packing them into a tuple, we would not like to convert anything to QVariant or QVariantMap.
I expect the current serialization to be significantly faster than packing the data into a QVariantMap.

Reply:
The structs need to be treated generically in "local" mode.
The QVariant replaces the LocalIPCMessage payload, right?
In other words cloning (QVariant) replaces serialization (LocalIPCMessage payload).
And why on earth cloning an object should be slower than serializing it?
If that is the case in facelift, the only reason is the overhead of creating std::tuples when cloning a struct.

Reply:
> The structs need to be treated generically in "local" mode.
> The QVariant replaces the LocalIPCMessage payload, right?

The QVariantMap does.

> In other words cloning (QVariant) replaces serialization (LocalIPCMessage payload).

The creation and population of the QVariantMap is what needs to be compared against the serialization.
We are talking about a map where the keys are QStrings and the elements are QVariants. The insertion of one element in that map involves the hashing of the key string, the conversion of the value to QVariant, ....
On the receiver side, a lookup in the map is needed, some type checking, etc....
In comparison, serializing/deserializing the value to a QByteArray is fast, unless the value is a large container which supports implicit sharing. Do you have such containers in mind ?

> And why on earth cloning an object should be slower than serializing it?
> If that is the case in facelift, the only reason is the overhead of creating std::tuples when cloning a struct.

Reply:
All of the operations in creating a QVariantMap is negligible compared to creation of QVariant itself (backed by KPI) 
unless you're interface has over tens of thousands of properties inside!!!

Premature optimization is the root of all evil.

Reply:
I agree that premature optimization is evil, and the current code is not the result of a premature optimization. Using the same serialization code in the local IPC as we do in the DBus IPC was a way to easily and quickly enable the local IPC. If high performance had really been required, no serialization would have been used.

I don't really see any nice alternative to the usage of a QVariant here though. Using std::variant could be better since it does not require all the type registrations and overhead which QVariant requires, but it would probably result in a more complex code.